### ISSUE TITLE
Cannot read tiff files saved with tiff writer.

### ISSUE THREAD

**Issue by Hovden** [OpenChemistry/tomviz#156]
Data saved with the tiff writer cannot be read back into tomviz. It instead prompts the user with:
"A reader for filename.tiff could not be found. Please choose one:"
Additionally, the tiff writer does not appear in the list of options.

**Comment by Hovden** [comment]
It also crashes after.

**Comment by cryos** [comment]
I tested the round tripping and it was working, but have verified locally with ParaView master/tomviz master that there are issues with the TIFF writer. We will look into it, I am actually seeing it get stuck in a loop and never finish writing on Linux with a small TIFF volume.

**Comment by Hovden** [comment]
Ok. I was demoing the software last night with someone's data and we ran into trouble. We did a reconstruction on a tiltseries then saved the data and tried to reload it. I will try to reproduce it.

**Comment by Hovden** [comment]
(Note: I originally posted this comment to the wrong issue)

To reproduce:
1) Load tiltseries data that is included.
2) Run the reconstruction.
3) Save data
4) Load data

It ends with a crash.

**Comment by mathturtle** [comment]
When I try to write the file I get errors printed out by the writer:

ERROR: In /home/shawn/dev/tomviz/src/paraview/VTK/IO/Image/vtkTIFFWriter.cxx, line 139
vtkTIFFWriter (0x8b19940): Unsupported data type: double

ERROR: In /home/shawn/dev/tomviz/src/paraview/VTK/IO/Image/vtkTIFFWriter.cxx, line 257
vtkTIFFWriter (0x8b19940): Problem writing file.

ERROR: In /home/shawn/dev/tomviz/src/paraview/VTK/IO/Image/vtkTIFFWriter.cxx, line 412
vtkTIFFWriter (0x8b19940): Problem writing trailer.

@cryos do you know anything about these?  Are they the cause of the bad file being written?

**Comment by cryos** [comment]
@mathturtle most certainly - the TIFF format does not support doubles. We should do some checks before attempting to write the TIFF - is it a supported type, if not can we convert it. Also, can we ensure the Python code produces either a float, or possibly a type matched with the input.

**Comment by mathturtle** [comment]
#165 addresses part of the problem here, disallowing double precision data from being written to tiff files.  The tiff writer issues that Marcus was seeing have a [fix being merged into VTK](https://gitlab.kitware.com/vtk/vtk/merge_requests/542) (this is taking slightly longer than normal since we want the fix in the current VTK release too, but should be merged to VTK tomorrow).

**Comment by cryos** [comment]
The patch was merged, but ParaView's VTK has not yet moved forward, we need at least commit 42e4eb0 from VTK for the TIFF fix.

**Comment by cryos** [comment]
We are waiting on [a ParaView merge request](https://gitlab.kitware.com/paraview/paraview/merge_requests/327) to push VTK forward in ParaView.

**Comment by cryos** [comment]
I just pushed the ParaView SHA forward, and we will have to wait for updated binaries to test. I think we can safely close this issue once those come back and we verify the TIFF issues are resolved. I think @Hovden already verified the main issue was resolved.

**Comment by Hovden** [comment]
My tiff writer is crashing using the test above (load tiltser, reconstruct, save).

Are we still waiting for VTK and ParaView?

**Comment by Hovden** [comment]
This works, but the system freezes up as tiff writer takes nearly two minutes to save a 256^3 dataset.

**Comment by mathturtle** [comment]
That is the bug that we were pushing PV forward to fix... Hmmm.

**Comment by cryos** [comment]
I just used the latest binary on my Mac and it wrote a 688^3 in a few seconds.

**Comment by Hovden** [comment]
Try loading the demo tilt series (256x256x74) then running "Reconstruct (Direct Fourier)" and then saving the data as a tiff.

**Comment by cryos** [comment]
Did that, took less than a second here using binary with 0.6.0-19-g6f3cd30 on a 2010 Mac running pretty recently updated OS.

**Comment by Hovden** [comment]
My mistake, spotlight was launching an older version.

Read / Write tiff is very fast in the latest build.

**Comment by cryos** [comment]
Great, thanks for verifying Rob. Closing this issue out as I think it is fully resolved now.
